instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/OtelResourceAutoConfiguration.java

```diff
@@ -42,27 +43,16 @@ public ResourceProvider otelEnvironmentResourceProvider() {
   }
 
   @Bean
-  public ResourceProvider otelSpringResourceProvider() {
-    return new SpringResourceProvider();
+  public ResourceProvider otelSpringResourceProvider(
```

Comment:
I hope that directly injecting `BuildProperties` doesn't have any weird ordering effects where it sometimes isn't available because `ProjectInfoAutoConfiguration` runs later. Also I hope that this does not cause any warnings similar to `2024-02-20 23:07:11.117 [INFO ] [main] [o.s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker] [PostProcessorRegistrationDelegate$BeanPostProcessorChecker.postProcessAfterInitialization:376] [] [] - Bean 'otel.traces.sampler-io.opentelemetry.instrumentation.spring.autoconfigure.SamplerProperties' of type [io.opentelemetry.instrumentation.spring.autoconfigure.SamplerProperties] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)` It would be nice if you could try this out on any spring boot app and see that you don't get these.

Reply:
good point - checked manually and it looks good.

Reply:
thanks